data/pools/transactionPool.go

```diff
@@ -698,15 +718,21 @@ func (pool *TransactionPool) recomputeBlockEvaluator(committedTxIds map[transact
 	}
 
 	pool.assemblyMu.Lock()
+	if (pool.assemblyDeadline != time.Time{}) {
+		asmStats.TransactionsLoopStartTime = int64(firstTxnGrpTime.Sub(pool.assemblyDeadline.Add(-config.ProposalAssemblyTime)))
```

Comment:
why `Add(-config.ProposalAssemblyTime)` and not `Sub` ?

Reply:
Because `Add` returns `time.Time` whereas `Sub` returns `time.Duration`

Reply:
Time.Add(Duration) -> Time
Time.Sub(Time) -> Duration

Reply:
It took me a bit to figure out that this is (deadline - allowedTime) to get the implicit start time, and then (first txn grp time - start time) shows how long the setup time to get there is. Dunno if there's a short comment that would clarify this intent and reasoning.